salt/cloud/clouds/vultrpy.py

```diff
@@ -132,13 +156,59 @@ def avail_scripts(conn=None):
     return _query("startupscript/list")
 
 
+def avail_firewall_groups(conn=None):
+    '''
+    return available firewall groups
+    '''
+    return _query('firewall/group_list')
+
+
+def avail_keys(conn=None):
+    '''
+    return available SSH keys
+    '''
+    return _query('sshkey/list')
+
+
 def list_scripts(conn=None, call=None):
     """
     return list of Startup Scripts
     """
     return avail_scripts()
 
 
+def list_firewall_groups(conn=None, call=None):
+    '''
+    return list of firewall groups
+    '''
+    return avail_firewall_groups()
+
+
+def list_keypairs(conn=None, call=None):
+    '''
+    return list of SSH keys
+    '''
+    return avail_keys()
+
+
+def show_keypair(kwargs=None, call=None):
+    '''
+    return list of SSH keys
+    '''
+    if not kwargs:
+        kwargs = {}
+
+    if 'keyname' not in kwargs:
+        log.error('A keyname is required.')
+        return False
+
+    keys = list_keypairs(call='function')
+    keyid = keys[kwargs['keyname']]['SSHKEYID']
+    log.debug('Key ID is %s', keyid)
```

Comment:
I do not have a vultr cloud instance to test this against. What information would be included here? I just want to ensure we are not logging sensitive information.

Reply:
This is the unique ID that Vultr sets for the key. It's used in the URLs when editing the key, such as `https://my.vultr.com/sshkeys/manage/?SSHKEYID={id}` so I don't believe it's sensitive.